src/coreclr/jit/hwintrinsicxarch.cpp

```diff
@@ -1415,15 +1415,192 @@ GenTree* Compiler::impSpecialIntrinsic(NamedIntrinsic        intrinsic,
 
         case NI_Vector128_ConvertToDouble:
         case NI_Vector256_ConvertToDouble:
+        case NI_Vector512_ConvertToDouble:
+        {
+            assert(sig->numArgs == 1);
+            assert(varTypeIsLong(simdBaseType) || simdBaseType == TYP_FLOAT);
+            if (IsBaselineVector512IsaSupportedOpportunistically())
+            {
+                if (varTypeIsLong(simdBaseType))
+                {
+                    intrinsic = (simdSize == 16) ? NI_AVX512DQ_VL_ConvertToVector128Double
+                              : (simdSize == 32) ? NI_AVX512DQ_VL_ConvertToVector256Double
+                              : NI_AVX512DQ_ConvertToVector512Double;
+                }
+                else
+                {
+                    intrinsic = (simdSize == 16) ? NI_SSE2_ConvertToVector128Double
+                              : (simdSize == 32) ? NI_AVX_ConvertToVector256Double
+                              : NI_AVX512F_ConvertToVector512Double;
+                }
+
+                op1     = impSIMDPopStack();
+                retNode = gtNewSimdHWIntrinsicNode(retType, op1, intrinsic, simdBaseJitType, simdSize);
+            }
+            break;
+        }
+
         case NI_Vector128_ConvertToInt64:
         case NI_Vector256_ConvertToInt64:
+        case NI_Vector512_ConvertToInt64:
+        {
+            assert(sig->numArgs == 1);
+            assert(simdBaseType == TYP_DOUBLE);
+#ifdef TARGET_AMD64
+            if (IsBaselineVector512IsaSupportedOpportunistically())
+            {
+                op1     = impSIMDPopStack();
+
+                var_types simdType = getSIMDTypeForSize(simdSize);
+                // Generate the control table for VFIXUPIMMSD
+                // The behavior we want is to saturate negative values to 0.
+                GenTreeVecCon* tbl = gtNewVconNode(simdType);
+
+                // QNAN: 0b1000: Saturate to Zero
+                // SNAN: 0b1000: Saturate to Zero
+                // ZERO: 0b0000
+                // +ONE: 0b0000
+                // -INF: 0b0000
+                // +INF: 0b0000
+                // -VAL: 0b0000
+                // +VAL: 0b0000
+                for ( int i = 0; i < 8; i++ )
+                {
+                    tbl->gtSimdVal.i64[i] = 0x00000088;
+                }
+
+                // Generate first operand
+                // The logic is that first and second operand are basically the same because we want 
+                // the output to be in the same xmm register
+                // Hence we clone the first operand
+                GenTree* op2Clone = fgMakeMultiUse(&op1);
+                // GenTree* op2Clone;
+                // op1 = impCloneExpr(op1, &op2Clone, CHECK_SPILL_ALL,
+                //                     nullptr DEBUGARG("Cloning double for Dbl2Ulng conversion"));
```

Comment:
Leftover comment that should be removed?

Reply:
*removed